stackslib/src/chainstate/stacks/db/mod.rs

```diff
@@ -298,12 +298,12 @@ impl DBConfig {
         });
         match epoch_id {
             StacksEpochId::Epoch10 => true,
-            StacksEpochId::Epoch20 => version_u32 >= 1 && version_u32 <= 5,
-            StacksEpochId::Epoch2_05 => version_u32 >= 2 && version_u32 <= 5,
-            StacksEpochId::Epoch21 => version_u32 >= 3 && version_u32 <= 5,
-            StacksEpochId::Epoch22 => version_u32 >= 3 && version_u32 <= 5,
-            StacksEpochId::Epoch23 => version_u32 >= 3 && version_u32 <= 5,
-            StacksEpochId::Epoch24 => version_u32 >= 3 && version_u32 <= 5,
+            StacksEpochId::Epoch20 => version_u32 >= 1 && version_u32 <= 6,
+            StacksEpochId::Epoch2_05 => version_u32 >= 2 && version_u32 <= 6,
+            StacksEpochId::Epoch21 => version_u32 >= 3 && version_u32 <= 6,
+            StacksEpochId::Epoch22 => version_u32 >= 3 && version_u32 <= 6,
+            StacksEpochId::Epoch23 => version_u32 >= 3 && version_u32 <= 6,
+            StacksEpochId::Epoch24 => version_u32 >= 3 && version_u32 <= 6,
             StacksEpochId::Epoch25 => version_u32 >= 3 && version_u32 <= 6,
             StacksEpochId::Epoch30 => version_u32 >= 3 && version_u32 <= 6,
```

Comment:
This is okay, but should we just drop these upper bounds? Realistically, if there was a chainstate release that had this upper bounding property, I think that would be a dramatic enough change that we would be adding other checks anyways, this is way more likely to be a footgun in the other direction (i.e., forgetting to expand the upper bound on the earlier epochs).

Reply:
I think this won't be a footgun in practice, because if we forget to update this check, we'll notice it right away since all the integration tests will panic.  This is because the migration logic (which calls this function) gets invoked in the Neon runloop on node bootup.

The rationale for keeping this check is that we won't accidentally ship a version of the node that corrupts existing chainstate.  Like, if a user starts a Stacks node that uses chainstate version 7, and then tries to start an older Stacks node that can only support up to 6 (e.g. suppose they revert), we'd _want_ that older node to panic.